lib/derivatives.rb

```diff
@@ -7,35 +7,35 @@ module PostZephirProcessing
   # `earliest_missing_date` is the main entrypoint when constructing an agenda of Zephir
   # file dates to fetch for processing.
   class Derivatives
+    # Location data for the derivatives we care about when constructing our list of missing dates.
+    # A file that has multiple locations (rights vs rights_archive) need only exist in one
+    # of them.
+    # TODO: is it even necessary to consider RIGHTS_DIR any more? Rights files can be expected to
```

Comment:
I think that's probably right, in which case you could back out the arrayification...